spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/DynamicConformanceJob.scala

```diff
@@ -230,17 +238,29 @@ object DynamicConformanceJob {
       .setBroadcastStrategyMode(broadcastingStrategyMode)
       .setBroadcastMaxSizeMb(broadcastingMaxSizeMb)
 
-    try {
+    Try {
       DynamicInterpreter.interpret(conformance, inputData)
-    } catch {
-      case e: ValidationException =>
+    } match {
+      case Failure(e: ValidationException) =>
         AtumImplicits.SparkSessionWrapper(spark).setControlMeasurementError("Conformance", e.getMessage, e.techDetails)
         throw e
-      case NonFatal(e)            =>
+      case Failure(NonFatal(e)) =>
         val sw = new StringWriter
         e.printStackTrace(new PrintWriter(sw))
         AtumImplicits.SparkSessionWrapper(spark).setControlMeasurementError("Conformance", e.getMessage, sw.toString)
         throw e
+      case Success(conformedDF) =>
+        if (SchemaUtils.fieldExists(Constants.EnceladusRecordId, conformedDF.schema)) {
+          conformedDF // no new id regenereration
+        } else {
+          recordIdGenerationStrategy match {
+            case UuidType.NoUuids => log.info("Record id generation is off.")
+            case UuidType.PseudoUuids => log.info("Record id generation is set to 'pseudo' - all runs will yield the same IDs.")
+            case UuidType.TrueUuids => log.info("Record id generation is on and true UUIDs will be added to output.")
```

Comment:
This could be part of `RecordIdGeneration.addRecordIdColumnByStrategy`

Reply:
Moved in the method, thanks.